mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/mail/JPAMapperProvider.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.james.mailbox.jpa.mail.model.JPAMailboxAnnotation;
+import org.apache.james.mailbox.jpa.mail.model.JPAProperty;
+import org.apache.james.mailbox.jpa.mail.model.JPAUserFlag;
+import org.apache.james.mailbox.jpa.mail.model.openjpa.AbstractJPAMailboxMessage;
+import org.apache.james.mailbox.jpa.mail.model.openjpa.JPAMailboxMessage;
+import org.apache.james.mailbox.jpa.user.model.JPASubscription;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.store.mail.AnnotationMapper;
+import org.apache.james.mailbox.store.mail.AttachmentMapper;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.MessageMapper;
+import org.apache.james.mailbox.store.mail.model.DefaultMessageId;
+import org.apache.james.mailbox.store.mail.model.MapperProvider;
+import org.apache.openjpa.persistence.OpenJPAPersistence;
+
+public class JPAMapperProvider implements MapperProvider {
+
+    @Override
+    public MailboxMapper createMailboxMapper() throws MailboxException {
+        return null;
+    }
+
+    @Override
+    public MessageMapper createMessageMapper() throws MailboxException {
+        return null;
+    }
+
+    @Override
+    public AttachmentMapper createAttachmentMapper() throws MailboxException {
+        return null;
+    }
+
+    @Override
+    public AnnotationMapper createAnnotationMapper() throws MailboxException {
+        return new TransactionalAnnotationMapper(new JPAAnnotationMapper(createEntityManagerFactory()));
+    }
+
+    @Override
+    public MailboxId generateId() {
+        return JPAId.of(Math.abs(new Random().nextInt()));
+    }
+
+    @Override
+    public MessageId generateMessageId() {
+        return new DefaultMessageId.Factory().generate();
+    }
+
+    @Override
+    public void clearMapper() throws MailboxException {
+        EntityManager entityManager = createEntityManagerFactory().createEntityManager();
+        entityManager.getTransaction().begin();
+        entityManager.createNativeQuery("TRUNCATE table JAMES_MAIL_USERFLAG;");
+        entityManager.createNativeQuery("TRUNCATE table JAMES_MAIL_PROPERTY;");
+        entityManager.createNativeQuery("TRUNCATE table JAMES_MAILBOX_ANNOTATION;");
+        entityManager.createNativeQuery("TRUNCATE table JAMES_MAILBOX;");
+        entityManager.createNativeQuery("TRUNCATE table JAMES_MAIL;");
+        entityManager.getTransaction().commit();
+        entityManager.close();
+    }
+
+    @Override
+    public void ensureMapperPrepared() throws MailboxException {
+
+    }
+
+    @Override
+    public boolean supportPartialAttachmentFetch() {
+        return false;
+    }
+
+    private EntityManagerFactory createEntityManagerFactory() {
+        HashMap<String, String> properties = new HashMap<String, String>();
+        properties.put("openjpa.ConnectionDriverName", "org.h2.Driver");
```

Comment:
Could we load from properties file? It should be easier for administrator if they would like to change DB